pkg/sysregistriesv2/system_registries_v2.go

```diff
@@ -519,6 +519,75 @@ func GetRegistries(ctx *types.SystemContext) ([]Registry, error) {
 	return config.Registries, nil
 }
 
+// LocationsForReferences returns a sorted list of locations (i.e., registries)
+// that may be contacted when pulling the specified references.  If the
+// references are empty, all unqualified-search registries and their mirrors
+// will be returned.
+func LocationsForReferences(ctx *types.SystemContext, references []string) ([]string, error) {
+	locations := make(map[string]bool)
+	needUnqualified := len(references) == 0
```

Comment:
Isn’t this just a caller error? Is it necessary to provide this? (The use case seems to be some version of “We may use unqualified names but don’t know which ones”, but that doesn’t work for mirrors of individual namespaces on the search registries.)